modules/renter/repair.go

```diff
@@ -352,6 +343,11 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 // managedDownloadChunkData downloads the requested chunk from Sia, for use in
 // the repair loop.
 func (r *Renter) managedDownloadChunkData(rs *repairState, file *file, offset uint64, chunkIndex uint64, chunkID chunkID) ([]byte, error) {
+	// Don't initiate too many downloads to avoid using up all memory
+	if len(rs.downloadingChunks) >= maxScheduledDownloads {
+		return nil, nil
+	}
```

Comment:
I don't think we have any tests that try to repair 5 chunks at the same time, so though the test suite is passing, this construction is not being verified at all.